### PR TITLE
arch/risc-v/src/mpfs/mpfs_i2c.c: Clean up using priv->status and STOP…

### PR DESCRIPTION
**Pull request by jlaitine**
… interrupts

- There are occasional extra STOPs being sent due to an IP bug when using an FPGA based I2C. Add a flag "inflight" to mask out extra STOP interrupts when using the FPGA based implementation
- There are no MPFS_I2C_ST_STOP_SENT irq's "initally". It is just already either success or still in progress

## Summary

This fixes some issues in barometer initialization at boot

## Testing

On-table

### CODE REVIEW

File: arch/risc-v/src/mpfs/mpfs_i2c.c [hunk 1]
```diff
@@ -566,12 +570,33 @@ static int mpfs_i2c_irq(int cpuint, void *context, void *arg)
...
         /* No activity, bus idle */
 
         break;
 
+      case MPFS_I2C_ST_STOP_SENT:
+
+        /* FPGA driver terminates all transactions with STOP sent irq
+         * if there has been no errors, the transfer succeeded.
+         * Due to the IP bug that extra data & STOPs can be sent after
+         * the actual transaction, filter out any extra stops wit
```
Comment (eenurkka) [inline arch/risc-v/src/mpfs/mpfs_i2c.c:583]: wit -> with

File: arch/risc-v/src/mpfs/mpfs_i2c.c [hunk 2]
```diff
@@ -477,6 +480,7 @@ static int mpfs_i2c_irq(int cpuint, void *context, void *arg)
 
             modifyreg32(MPFS_I2C_CTRL, MPFS_I2C_CTRL_SI_MASK, 0);
             clear_irq = 0u;
+            priv->inflight = false;
```
Comment (jlaitine) [inline arch/risc-v/src/mpfs/mpfs_i2c.c:483]: This is actually not needed

**Review by eenurkka — APPROVED** [review]
Looks good, only testing well will give the final verdict

**Review by pussuw — APPROVED** [review]
Lgtm

**Comment by jlaitine** [comment]
Positive test results, seems to also fix the issue where "reboot" command gets stuck at i2c devices sometimes for long time(out)s. Merging and up-streaming

**Pull request merged**